src/mlpack/methods/ann/ffn.hpp

```diff
@@ -279,44 +278,38 @@ class FFN
   template <class LayerType, class... Args>
   void Add(Args... args) { network.push_back(new LayerType(args...)); }
 
-  /**
-   * Add a new module to the model.
-   *
-   * @param layer The Layer to be added to the model.
-   */
-  void Add(Layer<arma::mat, arma::mat>* layer) { network.push_back(layer); }
-
   //! Get the network model.
-  const std::vector<Layer<arma::mat, arma::mat>*>& Model() const
+  const std::vector<Layer<InputType, OutputType>*>& Model() const
   {
     return network;
   }
-  //! Modify the network model.  Be careful!  If you change the structure of the
-  //! network or parameters for layers, its state may become invalid, so be sure
-  //! to call ResetParameters() afterwards.
-  //std::vector<Layer>& Model() { return network; }
+
+  /**
+   * Modify the network model.  Be careful!  If you change the structure of the
+   * network or parameters for layers, its state may become invalid, so be sure
+   * to call ResetParameters() afterwards.
+   */
+  std::vector<Layer<InputType, OutputType>>& Model() { return network; }
```

Comment:
Probably it should be
```cpp
std::vector<Layer<InputType, OutputType>*>& Model() { return network; }
```

Reply:
Right now we push an instantiated layer to the network, but a pointer would work as well, but that is easy to change, somewhat depends on the serialization part what would be easier.

Reply:
I think we are dynamically allocating memory when we use `model->Add<LayerType>(args...)`.
```cpp
template <class LayerType, class... Args>
void Add(Args... args) { network.push_back(new LayerType(args...)); }
```
...and also the data type of `network` is `std::vector<Layer<InputType, OutputType>* >`.

Reply:
You are right, I missed that part, so yes, that should be `std::vector<Layer<InputType, OutputType>*>& Model() { return network; }`.